clang/test/SemaCXX/unsafe-buffer-usage-pragma.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clang_cc1 -std=c++17 -Wno-all -Wunsafe-buffer-usage \
+// RUN:            -verify %s
+
+// Previously we had a bug where we would diagnose *no* unsafe buffer warnings
+// if the warning was disabled by pragma and left disabled until end-of-TU.
+// This is a reasonable way to disable unsafe buffer warnings on an entire
+// .c/cpp file, and it shouldn't disable the warnings in headers or previous
+// source locations, so we test that this works.
+
+// FIXME: This RUNX line should pass, but it does not, because we check if the
```

Comment:
This is unfortunate, but enabling warnings with pragmas seems much less common than ignoring them, so if we have to choose this seems better.